clickhouse-http-client/src/main/java/com/clickhouse/client/http/ClickHouseHttpConnection.java

```diff
@@ -358,6 +388,11 @@ protected ClickHouseHttpConnection(ClickHouseNode server, ClickHouseRequest<?> r
             throw new IllegalArgumentException("Non-null server and request are required");
         }
 
+        synchronized (ClickHouseHttpConnection.class) {
```

Comment:
To avoid extra work please move local address logic into `com.clickhouse.client.http.ClickHouseHttpClient`
1. do initialization in the `ClickHouseHttpClient` constructor and instance fields to keep address and host name. 
2. add extra parameter `referer` to the `com.clickhouse.client.http.ClickHouseHttpConnection#createDefaultHeaders` (we will refactor it later) and path it from client where used. If referer == null or empty - no Referer header should be sent.